main/src/ca/uwaterloo/flix/language/phase/sjvm/BytecodeCompiler.scala

```diff
@@ -53,9 +140,145 @@ object BytecodeCompiler {
 
   def pushInt[R <: Stack](n: Int): F[R] => F[R ** PrimInt32] = ???
 
+  def pushChar[R <: Stack](c: Char): F[R] => F[R ** PrimChar] = ???
+
+  object ArrayInstructions {
+
+    def arrayLength[R <: Stack](): F[R ** JArray[JType]] => F[R ** PrimInt32] = ???
+
+    def BALoad[R <: Stack](): F[R ** JArray[PrimInt8] ** PrimInt32] => F[R ** PrimInt8] = ???
+
+    def SALoad[R <: Stack](): F[R ** JArray[PrimInt16] ** PrimInt32] => F[R ** PrimInt16] = ???
+
+    def IALoad[R <: Stack](): F[R ** JArray[PrimInt32] ** PrimInt32] => F[R ** PrimInt32] = ???
+
+    def LALoad[R <: Stack](): F[R ** JArray[PrimInt64] ** PrimInt32] => F[R ** PrimInt64] = ???
+
+    def CALoad[R <: Stack](): F[R ** JArray[PrimChar] ** PrimInt32] => F[R ** PrimChar] = ???
+
+    def FALoad[R <: Stack](): F[R ** JArray[PrimFloat32] ** PrimInt32] => F[R ** PrimFloat32] = ???
+
+    def DALoad[R <: Stack](): F[R ** JArray[PrimFloat64] ** PrimInt32] => F[R ** PrimFloat64] = ???
+
+    def AALoad[R <: Stack](): F[R ** JArray[JObject] ** PrimInt32] => F[R ** JObject] = ???
+
+    def XALoad[R <: Stack, T <: JType](tpe: ErasedType[T]): F[R ** JArray[T] ** PrimInt32] => F[R ** T] = tpe match {
+      case ErasedType.Unit() => ???
+      case ErasedType.Bool() => ???
+      case ErasedType.BoxedBool() => ???
+      case ErasedType.Char() => CALoad()
+      case ErasedType.BoxedChar() => ???
+      case ErasedType.Float32() => ???
+      case ErasedType.BoxedFloat32() => ???
+      case ErasedType.Float64() => ???
+      case ErasedType.BoxedFloat64() => ???
+      case ErasedType.Int8() => BALoad()
+      case ErasedType.BoxedInt8() => ???
+      case ErasedType.Int16() => SALoad()
+      case ErasedType.BoxedInt16() => ???
+      case ErasedType.Int32() => IALoad()
+      case ErasedType.BoxedInt32() => ???
+      case ErasedType.Int64() => LALoad()
+      case ErasedType.BoxedInt64() => ???
+      case ErasedType.Array(tpe) => ???
+      case ErasedType.Channel(tpe) => ???
+      case ErasedType.Lazy(tpe) => ???
+      case ErasedType.Ref(tpe) => ???
+      case _: ErasedType.Str | _: ErasedType.BigInt | _: ErasedType.Tuple | _: ErasedType.Enum | _: ErasedType.Arrow |
```

Comment:
What would happen if we changed ErasedType:

```
sealed trait ErasedType[...]
object ErasedType {
  case class Int32() extends ErasedType[...]
  case class Float32() extends ErasedType[...]
  case class Obj(specificType: SpecificType) extends ErasedType[JObject]
}

sealed trait SpecificType
object SpecificType {
  case class Tuple ...
  case class Enum ...
}
```

and so forth? Is this helpful or not?

Reply:
Yeah thatd make many things shorter, have a generel object pile of types.